sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/DataSourceV2AnalysisSuite.scala

```diff
@@ -290,15 +290,14 @@ abstract class DataSourceV2AnalysisSuite extends AnalysisTest {
       StructField("y", DoubleType))).toAttributes)
 
     val query = TestRelation(StructType(Seq(
-      StructField("x", DoubleType),
+      StructField("x", FloatType),
```

Comment:
why this change?

Reply:
In this test case, the error message `"Cannot safely cast", "'x'", "DoubleType to FloatType"` is gone after code changes.
So I think we should make the test case simpler. The nullability error in the `x` column is enough.

Reply:
No, all errors should be reported. Attempting to write a double to a float column is an error that should be shown.